src/mugs.js

```diff
@@ -295,7 +295,45 @@ define([
             rawControlXML: {
                 presence: 'optional',
                 lstring: 'Raw XML'
-            }
+            },
+            dataParent: {
+                lstring: 'Data Parent',
+                visibility: function(mug) {
+                    function recFunc(mug) {
+                        if (!mug) {
+                            return true;
+                        } else if (mug.__className === 'Repeat' ||
+                                   mug.__className === "Item") {
+                            return false;
+                        }
+                        return recFunc(mug.form.tree.getNodeFromMug(mug).parent.value);
```

Comment:
If you're changing the code above, you could change this to `recFunc(mug.parentMug)` since `mug.parentMug` is now a calculated property that does what this is doing.